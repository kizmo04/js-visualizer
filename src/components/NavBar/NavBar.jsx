import React, { Component } from "react";
import styles from "./NavBar.module.scss";

class NavBar extends Component {
  constructor(props) {
    super(props);
    this.handleInputRangeChange = this.handleInputRangeChange.bind(this);
  }
  handleInputRangeChange (e) {
    this.props.onRunningSpeedChange(e.target.value);
  }
  render() {
    const { isRunning, onStepButtonClick, onRunButtonClick, onRestartButtonClick, onShareButtonClick } = this.props;
    return (
      <nav className={`${styles.backgroundTransparent} navbar`}>
        <div className="navbar-brand">
          <div className="navbar-item">VisualCode.Studio</div>
          <div className="navbar-start">
            <div className="navbar-item">
              <div className="field is-grouped">
                <div
                  className={`${styles.marginRight} button is-small is-info step-button`}
                  onClick={onStepButtonClick}
                >
                  Next Step
                </div>
                <div
                  className={`${styles.marginRight} button is-small is-info run-button`}
                  onClick={onRunButtonClick}
                >
                  {isRunning ? "Stop" : "Run"}
                </div>
                <input
                  className={`${
                    styles.marginRight
                  } slider is-fullwidth is-info running-speed-slider`}
                  step="1"
                  min="0"
                  max="100"
                  defaultValue="50"
                  type="range"
                  onChange={this.handleInputRangeChange}
                />
                <div
                  className={`${styles.marginNone} button is-small is-info restart-button`}
                  onClick={onRestartButtonClick}
                >
                  Restart
                </div>
              </div>
            </div>
          </div>
          <div className="navbar-end">
            <div className="navbar-item">
              <div
                className="button is-info is-small share-button"
                onClick={onShareButtonClick}
              >
                Share
              </div>
            </div>
          </div>
        </div>
        <div className="navbar-menu">
        </div>
      </nav>
    );
  }
}

export default NavBar;
