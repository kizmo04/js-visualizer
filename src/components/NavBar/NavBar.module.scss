.marginRight {
  margin: 0 1rem 0 0;
}

.marginNone {
  margin: 0;
}

.paddingNone {
  padding: 0;
}

.backgroundTransparent {
  background-color: transparent;
}

.icon {
  font-size: 1.2rem;
  vertical-align: middle;
}