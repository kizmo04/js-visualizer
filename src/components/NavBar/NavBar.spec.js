import React from "react";
import { shallow } from "enzyme";
import NavBar from "./NavBar";

const setup = (setupProps = {}) => {
  const defaultProps = {
    isRunning: false,
    hasNextStep: true,
    runningSpeed: 100,
    code: "",
    onStepButtonClick: jest.fn(),
    onShareButtonClick: jest.fn(),
    onRestartButtonClick: jest.fn(),
    onRunButtonClick: jest.fn(),
    onRunningSpeedChange: jest.fn()
  };
  const prototypeMethods = {
    onChangeSpy: jest.spyOn(NavBar.prototype, "handleInputRangeChange")
  };
  const props = { ...defaultProps, ...setupProps };
  const component = shallow(<NavBar {...props} />);

  return {
    component,
    props,
    prototypeMethods
  };
};

describe("NavBar", () => {
  it("renders without crashing", () => {
    const { component } = setup();

    expect(component).toMatchSnapshot();
  });

  describe("with props", () => {
    it("first render run button should contain 'run'", () => {
      const { component } = setup();

      expect(component.find(".run-button").text()).toBe("Run");
    });

    it("render run button to display 'stop' if isRunning is false", () => {
      const { component } = setup({
        isRunning: true
      });

      expect(component.find(".run-button").text()).toBe("Stop");
    });
  });

  describe("events", () => {
    it("onClick calls onRunButtonClick", () => {
      const { component, props } = setup();

      component.find(".run-button").simulate("click");
      expect(props.onRunButtonClick).toHaveBeenCalled();
    });

    it("onChange calls onRunningSpeedChange", () => {
      const { component, props, prototypeMethods } = setup();
      const e = { target: { value: 50 } };

      component.update();
      component.find(".running-speed-slider").simulate("change", e);

      expect(prototypeMethods.onChangeSpy).toHaveBeenCalledWith(e);
      expect(props.onRunningSpeedChange).toHaveBeenCalledWith(50);
    });

    it("onClick calls onRestartButtonClick", () => {
      const { component, props } = setup();

      component.find(".restart-button").simulate("click");
      expect(props.onRestartButtonClick).toHaveBeenCalled();
    });

    it("onClick calls onShareButtonClick", () => {
      const { component, props } = setup();

      component.find(".share-button").simulate("click");
      expect(props.onShareButtonClick).toHaveBeenCalled();
    });

    it("onClick calls onStepButtonClick", () => {
      const { component, props } = setup();

      component.find(".step-button").simulate("click");
      expect(props.onStepButtonClick).toHaveBeenCalled();
    });
  });
});
