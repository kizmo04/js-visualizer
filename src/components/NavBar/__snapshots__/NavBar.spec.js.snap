// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavBar
    code=""
    hasNextStep={true}
    isRunning={false}
    onRestartButtonClick={[MockFunction]}
    onRunButtonClick={[MockFunction]}
    onRunningSpeedChange={[MockFunction]}
    onShareButtonClick={[MockFunction]}
    onStepButtonClick={[MockFunction]}
    runningSpeed={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="navbar-brand"
        >
          <div
            className="navbar-item"
          >
            VisualCode.Studio
          </div>
          <div
            className="navbar-start"
          >
            <div
              className="navbar-item"
            >
              <div
                className="field is-grouped"
              >
                <div
                  className="marginRight button is-small is-info step-button"
                  onClick={[MockFunction]}
                >
                  Next Step
                </div>
                <div
                  className="marginRight button is-small is-info run-button"
                  onClick={[MockFunction]}
                >
                  Run
                </div>
                <input
                  className="marginRight slider is-fullwidth is-info running-speed-slider"
                  defaultValue="50"
                  max="100"
                  min="0"
                  onChange={[Function]}
                  step="1"
                  type="range"
                />
                <div
                  className="marginNone button is-small is-info restart-button"
                  onClick={[MockFunction]}
                >
                  Restart
                </div>
              </div>
            </div>
          </div>
          <div
            className="navbar-end"
          >
            <div
              className="navbar-item"
            >
              <div
                className="button is-info is-small share-button"
                onClick={[MockFunction]}
              >
                Share
              </div>
            </div>
          </div>
        </div>,
        <div
          className="navbar-menu"
        />,
      ],
      "className": "backgroundTransparent navbar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="navbar-item"
            >
              VisualCode.Studio
            </div>,
            <div
              className="navbar-start"
            >
              <div
                className="navbar-item"
              >
                <div
                  className="field is-grouped"
                >
                  <div
                    className="marginRight button is-small is-info step-button"
                    onClick={[MockFunction]}
                  >
                    Next Step
                  </div>
                  <div
                    className="marginRight button is-small is-info run-button"
                    onClick={[MockFunction]}
                  >
                    Run
                  </div>
                  <input
                    className="marginRight slider is-fullwidth is-info running-speed-slider"
                    defaultValue="50"
                    max="100"
                    min="0"
                    onChange={[Function]}
                    step="1"
                    type="range"
                  />
                  <div
                    className="marginNone button is-small is-info restart-button"
                    onClick={[MockFunction]}
                  >
                    Restart
                  </div>
                </div>
              </div>
            </div>,
            <div
              className="navbar-end"
            >
              <div
                className="navbar-item"
              >
                <div
                  className="button is-info is-small share-button"
                  onClick={[MockFunction]}
                >
                  Share
                </div>
              </div>
            </div>,
          ],
          "className": "navbar-brand",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "VisualCode.Studio",
              "className": "navbar-item",
            },
            "ref": null,
            "rendered": "VisualCode.Studio",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="navbar-item"
              >
                <div
                  className="field is-grouped"
                >
                  <div
                    className="marginRight button is-small is-info step-button"
                    onClick={[MockFunction]}
                  >
                    Next Step
                  </div>
                  <div
                    className="marginRight button is-small is-info run-button"
                    onClick={[MockFunction]}
                  >
                    Run
                  </div>
                  <input
                    className="marginRight slider is-fullwidth is-info running-speed-slider"
                    defaultValue="50"
                    max="100"
                    min="0"
                    onChange={[Function]}
                    step="1"
                    type="range"
                  />
                  <div
                    className="marginNone button is-small is-info restart-button"
                    onClick={[MockFunction]}
                  >
                    Restart
                  </div>
                </div>
              </div>,
              "className": "navbar-start",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="field is-grouped"
                >
                  <div
                    className="marginRight button is-small is-info step-button"
                    onClick={[MockFunction]}
                  >
                    Next Step
                  </div>
                  <div
                    className="marginRight button is-small is-info run-button"
                    onClick={[MockFunction]}
                  >
                    Run
                  </div>
                  <input
                    className="marginRight slider is-fullwidth is-info running-speed-slider"
                    defaultValue="50"
                    max="100"
                    min="0"
                    onChange={[Function]}
                    step="1"
                    type="range"
                  />
                  <div
                    className="marginNone button is-small is-info restart-button"
                    onClick={[MockFunction]}
                  >
                    Restart
                  </div>
                </div>,
                "className": "navbar-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="marginRight button is-small is-info step-button"
                      onClick={[MockFunction]}
                    >
                      Next Step
                    </div>,
                    <div
                      className="marginRight button is-small is-info run-button"
                      onClick={[MockFunction]}
                    >
                      Run
                    </div>,
                    <input
                      className="marginRight slider is-fullwidth is-info running-speed-slider"
                      defaultValue="50"
                      max="100"
                      min="0"
                      onChange={[Function]}
                      step="1"
                      type="range"
                    />,
                    <div
                      className="marginNone button is-small is-info restart-button"
                      onClick={[MockFunction]}
                    >
                      Restart
                    </div>,
                  ],
                  "className": "field is-grouped",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Next Step",
                      "className": "marginRight button is-small is-info step-button",
                      "onClick": [MockFunction],
                    },
                    "ref": null,
                    "rendered": "Next Step",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Run",
                      "className": "marginRight button is-small is-info run-button",
                      "onClick": [MockFunction],
                    },
                    "ref": null,
                    "rendered": "Run",
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "marginRight slider is-fullwidth is-info running-speed-slider",
                      "defaultValue": "50",
                      "max": "100",
                      "min": "0",
                      "onChange": [Function],
                      "step": "1",
                      "type": "range",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Restart",
                      "className": "marginNone button is-small is-info restart-button",
                      "onClick": [MockFunction],
                    },
                    "ref": null,
                    "rendered": "Restart",
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="navbar-item"
              >
                <div
                  className="button is-info is-small share-button"
                  onClick={[MockFunction]}
                >
                  Share
                </div>
              </div>,
              "className": "navbar-end",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="button is-info is-small share-button"
                  onClick={[MockFunction]}
                >
                  Share
                </div>,
                "className": "navbar-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Share",
                  "className": "button is-info is-small share-button",
                  "onClick": [MockFunction],
                },
                "ref": null,
                "rendered": "Share",
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "navbar-menu",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="navbar-brand"
          >
            <div
              className="navbar-item"
            >
              VisualCode.Studio
            </div>
            <div
              className="navbar-start"
            >
              <div
                className="navbar-item"
              >
                <div
                  className="field is-grouped"
                >
                  <div
                    className="marginRight button is-small is-info step-button"
                    onClick={[MockFunction]}
                  >
                    Next Step
                  </div>
                  <div
                    className="marginRight button is-small is-info run-button"
                    onClick={[MockFunction]}
                  >
                    Run
                  </div>
                  <input
                    className="marginRight slider is-fullwidth is-info running-speed-slider"
                    defaultValue="50"
                    max="100"
                    min="0"
                    onChange={[Function]}
                    step="1"
                    type="range"
                  />
                  <div
                    className="marginNone button is-small is-info restart-button"
                    onClick={[MockFunction]}
                  >
                    Restart
                  </div>
                </div>
              </div>
            </div>
            <div
              className="navbar-end"
            >
              <div
                className="navbar-item"
              >
                <div
                  className="button is-info is-small share-button"
                  onClick={[MockFunction]}
                >
                  Share
                </div>
              </div>
            </div>
          </div>,
          <div
            className="navbar-menu"
          />,
        ],
        "className": "backgroundTransparent navbar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="navbar-item"
              >
                VisualCode.Studio
              </div>,
              <div
                className="navbar-start"
              >
                <div
                  className="navbar-item"
                >
                  <div
                    className="field is-grouped"
                  >
                    <div
                      className="marginRight button is-small is-info step-button"
                      onClick={[MockFunction]}
                    >
                      Next Step
                    </div>
                    <div
                      className="marginRight button is-small is-info run-button"
                      onClick={[MockFunction]}
                    >
                      Run
                    </div>
                    <input
                      className="marginRight slider is-fullwidth is-info running-speed-slider"
                      defaultValue="50"
                      max="100"
                      min="0"
                      onChange={[Function]}
                      step="1"
                      type="range"
                    />
                    <div
                      className="marginNone button is-small is-info restart-button"
                      onClick={[MockFunction]}
                    >
                      Restart
                    </div>
                  </div>
                </div>
              </div>,
              <div
                className="navbar-end"
              >
                <div
                  className="navbar-item"
                >
                  <div
                    className="button is-info is-small share-button"
                    onClick={[MockFunction]}
                  >
                    Share
                  </div>
                </div>
              </div>,
            ],
            "className": "navbar-brand",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "VisualCode.Studio",
                "className": "navbar-item",
              },
              "ref": null,
              "rendered": "VisualCode.Studio",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="navbar-item"
                >
                  <div
                    className="field is-grouped"
                  >
                    <div
                      className="marginRight button is-small is-info step-button"
                      onClick={[MockFunction]}
                    >
                      Next Step
                    </div>
                    <div
                      className="marginRight button is-small is-info run-button"
                      onClick={[MockFunction]}
                    >
                      Run
                    </div>
                    <input
                      className="marginRight slider is-fullwidth is-info running-speed-slider"
                      defaultValue="50"
                      max="100"
                      min="0"
                      onChange={[Function]}
                      step="1"
                      type="range"
                    />
                    <div
                      className="marginNone button is-small is-info restart-button"
                      onClick={[MockFunction]}
                    >
                      Restart
                    </div>
                  </div>
                </div>,
                "className": "navbar-start",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="field is-grouped"
                  >
                    <div
                      className="marginRight button is-small is-info step-button"
                      onClick={[MockFunction]}
                    >
                      Next Step
                    </div>
                    <div
                      className="marginRight button is-small is-info run-button"
                      onClick={[MockFunction]}
                    >
                      Run
                    </div>
                    <input
                      className="marginRight slider is-fullwidth is-info running-speed-slider"
                      defaultValue="50"
                      max="100"
                      min="0"
                      onChange={[Function]}
                      step="1"
                      type="range"
                    />
                    <div
                      className="marginNone button is-small is-info restart-button"
                      onClick={[MockFunction]}
                    >
                      Restart
                    </div>
                  </div>,
                  "className": "navbar-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        className="marginRight button is-small is-info step-button"
                        onClick={[MockFunction]}
                      >
                        Next Step
                      </div>,
                      <div
                        className="marginRight button is-small is-info run-button"
                        onClick={[MockFunction]}
                      >
                        Run
                      </div>,
                      <input
                        className="marginRight slider is-fullwidth is-info running-speed-slider"
                        defaultValue="50"
                        max="100"
                        min="0"
                        onChange={[Function]}
                        step="1"
                        type="range"
                      />,
                      <div
                        className="marginNone button is-small is-info restart-button"
                        onClick={[MockFunction]}
                      >
                        Restart
                      </div>,
                    ],
                    "className": "field is-grouped",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Next Step",
                        "className": "marginRight button is-small is-info step-button",
                        "onClick": [MockFunction],
                      },
                      "ref": null,
                      "rendered": "Next Step",
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Run",
                        "className": "marginRight button is-small is-info run-button",
                        "onClick": [MockFunction],
                      },
                      "ref": null,
                      "rendered": "Run",
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "marginRight slider is-fullwidth is-info running-speed-slider",
                        "defaultValue": "50",
                        "max": "100",
                        "min": "0",
                        "onChange": [Function],
                        "step": "1",
                        "type": "range",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Restart",
                        "className": "marginNone button is-small is-info restart-button",
                        "onClick": [MockFunction],
                      },
                      "ref": null,
                      "rendered": "Restart",
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="navbar-item"
                >
                  <div
                    className="button is-info is-small share-button"
                    onClick={[MockFunction]}
                  >
                    Share
                  </div>
                </div>,
                "className": "navbar-end",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="button is-info is-small share-button"
                    onClick={[MockFunction]}
                  >
                    Share
                  </div>,
                  "className": "navbar-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Share",
                    "className": "button is-info is-small share-button",
                    "onClick": [MockFunction],
                  },
                  "ref": null,
                  "rendered": "Share",
                  "type": "div",
                },
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "navbar-menu",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
